cdc/sinkv2/eventsink/cloudstorage/cloud_storage_dml_sink.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cloudstorage
+
+import (
+	"context"
+	"net/url"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/cdc/model"
+	rcommon "github.com/pingcap/tiflow/cdc/redo/common"
+	"github.com/pingcap/tiflow/cdc/sink/codec/builder"
+	"github.com/pingcap/tiflow/cdc/sink/codec/common"
+	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/cdc/sinkv2/util"
+	"github.com/pingcap/tiflow/pkg/chann"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/sink/cloudstorage"
+)
+
+const (
+	defaultEncodingConcurrency = 8
+	defaultMaxMessageBytes     = 1073741824 // 1GB
+)
+
+// Assert EventSink[E event.TableEvent] implementation
+var _ eventsink.EventSink[*model.SingleTableTxn] = (*sink)(nil)
+
+// versionedTable is used to wrap TableName with a version
+type versionedTable struct {
+	model.TableName
+	TableVersion uint64
+}
+
+// eventFragment is used to attach a sequence number to TxnCallbackableEvent.
+// The sequence number is mainly useful for TxnCallbackableEvent defragmentation.
+// e.g. TxnCallbackableEvent 1~5 are dispatched to a group of encoding workers, but the
+// encoding completion time varies. Let's say the final completion sequence are 1,3,2,5,4,
+// we can use the sequence numbers to do defragmentation so that the events can arrive
+// at dmlWorker sequentially.
+type eventFragment struct {
+	// event sequence number
+	seqNumber    uint64
+	tableVersion uint64
+	tableName    model.TableName
```

Comment:
it looks that L58 ~ L59 is duplicate of `versionedTable`

Reply:
Done.